src/plugins/discover/public/application/main/components/layout/discover_panels_resizable.tsx

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { EuiResizableContainer, useGeneratedHtmlId, useResizeObserver } from '@elastic/eui';
+import { css } from '@emotion/react';
+import React, { ReactElement, RefObject, useCallback, useEffect, useState } from 'react';
+
+const percentToPixels = (containerHeight: number, percentage: number) =>
+  Math.round(containerHeight * (percentage / 100));
+
+const pixelsToPercent = (containerHeight: number, pixels: number) =>
+  +((pixels / containerHeight) * 100).toFixed(4);
+
+export const DiscoverPanelsResizable = ({
+  className,
+  resizeRef,
+  initialTopPanelHeight,
+  minTopPanelHeight,
+  minMainPanelHeight,
+  topPanel,
+  mainPanel,
+}: {
+  className?: string;
+  resizeRef: RefObject<HTMLDivElement>;
+  initialTopPanelHeight: number;
+  minTopPanelHeight: number;
+  minMainPanelHeight: number;
+  topPanel: ReactElement;
+  mainPanel: ReactElement;
+}) => {
+  const topPanelId = useGeneratedHtmlId({ prefix: 'topPanel' });
+  const { height: containerHeight } = useResizeObserver(resizeRef.current);
+  const [topPanelHeight, setTopPanelHeight] = useState(initialTopPanelHeight);
+  const [panelSizes, setPanelSizes] = useState({ topPanelSize: 0, mainPanelSize: 0 });
+
+  // EuiResizableContainer doesn't work properly when used with react-reverse-portal and
+  // will cancel the resize. To work around this we keep track of when resizes start and
+  // end to toggle the rendering of a transparent overlay which prevents the cancellation.
+  // EUI issue: https://github.com/elastic/eui/issues/6199
+  const [resizeWithPortalsHackIsResizing, setResizeWithPortalsHackIsResizing] = useState(false);
```

Comment:
FYI we'll be able to make another PR to clean up a lot of the hackiness here once this EUI ticket is resolved: https://github.com/elastic/eui/issues/6225

Reply:
Issue created for this: https://github.com/elastic/kibana/issues/140542.